src/api/app/controllers/labels_controller.rb

```diff
@@ -0,0 +1,85 @@
+class LabelsController < ApplicationController
+  skip_before_action :require_login, only: [:index]
+  before_action :xml_hash, :find_labelable_and_label_template, only: [:create]
+
+  # GET /labels/:project/:package
+  # GET /labels/:number
+  def index
+    if params[:number]
+      labelable = BsRequest.find_by_number(params[:number])
+      message = "Unable to find request '#{params[:number]}'"
+    else
+      labelable = Package.find_by_project_and_name(params[:project], params[:package])
```

Comment:
We use `Package.get_by_project_and_name` to instantiate packages. See the other API controllers that deal with packages.

Reply:
Done. It has been updated